plugins/janus_streaming.c

```diff
@@ -1015,6 +1017,14 @@ struct janus_plugin_result *janus_streaming_handle_message(janus_plugin_session
 			json_object_set_new(ml, "id", json_integer(mp->id));
 			json_object_set_new(ml, "description", json_string(mp->description));
 			json_object_set_new(ml, "type", json_string(mp->streaming_type == janus_streaming_type_live ? "live" : "on demand"));
+			if(mp->streaming_source == janus_streaming_source_rtp) {
+				janus_streaming_rtp_source *source = mp->source;
+				gint64 now = janus_get_monotonic_time();
+				if ( source->audio_fd != -1 )
+					json_object_set_new(ml, "audio_age_ms", json_integer((now - source->last_received_audio) / 1000));
+				if ( source->video_fd != -1 )
+					json_object_set_new(ml, "video_age_ms", json_integer((now - source->last_received_video) / 1000));
```

Comment:
Just for our coding conventions, could you remove those extra spaces in the two ifs?

    if(source->audio_fd != -1)

For the rest, the patch looks fine to me and, apart from the considerations on whether or not we need locking (something we can fix along the way, if needed) we can merge it. Thanks for working on this!